modules/framework/galasa-parent/dev.galasa.framework.k8s.controller/src/main/java/dev/galasa/framework/k8s/controller/InterruptedRunEventProcessor.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.framework.k8s.controller;
+
+import java.util.List;
+import java.util.Queue;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import dev.galasa.framework.spi.DynamicStatusStoreException;
+import dev.galasa.framework.spi.IFrameworkRuns;
+import dev.galasa.framework.spi.IResultArchiveStore;
+import dev.galasa.framework.spi.IResultArchiveStoreDirectoryService;
+import dev.galasa.framework.spi.IRunResult;
+import dev.galasa.framework.spi.ResultArchiveStoreException;
+import dev.galasa.framework.spi.RunRasAction;
+import dev.galasa.framework.spi.teststructure.TestStructure;
+
+/**
+ * InterruptedRunEventProcessor runs as a thread in the engine controller pod and it maintains a
+ * queue of interrupt events.
+ * 
+ * When the interrupt event queue is not empty, it removes each event from the queue and processes them
+ * by marking interrupted runs as finished in the DSS and updating their RAS records as defined by the
+ * deferred RAS actions within the interrupt event.
+ */
+public class InterruptedRunEventProcessor implements Runnable {
+
+    private Log logger = LogFactory.getLog(getClass());
+
+    private final Queue<RunInterruptEvent> queue;
+    private IFrameworkRuns frameworkRuns;
+    private IResultArchiveStore rasStore;
+
+    public InterruptedRunEventProcessor(Queue<RunInterruptEvent> queue, IFrameworkRuns frameworkRuns, IResultArchiveStore rasStore) {
+        this.queue = queue;
+        this.frameworkRuns = frameworkRuns;
+        this.rasStore = rasStore;
+    }
+
```

Comment:
Gets called periodically, processes the entire queue of events, then exits.

Reply:
Javadoc comment added in https://github.com/galasa-dev/galasa/pull/273/commits/9268879a4e549b67260cb2fef9cb163d6c0ac963